source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -218,12 +384,20 @@ bool AggressiveDCEPass::AggressiveDCE(ir::Function* func) {
     }
     worklist_.pop();
   }
-  // Mark all non-live instructions dead
-  for (auto& blk : *func) {
-    for (auto& inst : blk) {
-      if (live_insts_.find(&inst) != live_insts_.end())
+  // Mark all non-live instructions dead, except non-if branches
+  for (auto bi = structuredOrder.begin(); bi != structuredOrder.end(); ++bi) {
+    for (auto ii = (*bi)->begin(); ii != (*bi)->end(); ++ii) {
+      if (IsLive(&*ii))
         continue;
-      dead_insts_.insert(&inst);
+      if (IsBranch(ii->opcode())) {
+        if (ii == (*bi)->begin())
+          continue;
+        auto mi = ii;
+        --mi;
+        if (mi->opcode() != SpvOpSelectionMerge)
```

Comment:
Is there a reason this will not work (or we don't want it to work) with branches that are not marked as OpSelectionMerge?  That reason should be documented as a comment.

Reply:
Specifically, OpSwitch is the other case.   Totally fine to not add it now since that's way more testing burden.  Agree it would be nice to have a comment saying you've deliberately scoped it down to just if-selection